cmd/zpool/zpool_main.c

```diff
@@ -9515,26 +9515,35 @@ static void
 errors_nvlist(zpool_handle_t *zhp, status_cbdata_t *cb, nvlist_t *item)
 {
 	uint64_t nerr;
+	int verbosity = cb->cb_verbosity;
 	nvlist_t *config = zpool_get_config(zhp, NULL);
 	if (nvlist_lookup_uint64(config, ZPOOL_CONFIG_ERRCOUNT,
 	    &nerr) == 0) {
 		nice_num_str_nvlist(item, ZPOOL_CONFIG_ERRCOUNT, nerr,
 		    cb->cb_literal, cb->cb_json_as_int, ZFS_NICENUM_1024);
-		if (nerr != 0 && cb->cb_verbose) {
+		if (nerr != 0 && cb->cb_verbosity > 0) {
 			nvlist_t *nverrlist = NULL;
-			if (zpool_get_errlog(zhp, &nverrlist) == 0) {
+			if (zpool_get_errlog(zhp, &nverrlist, verbosity) == 0) {
 				int i = 0;
 				int count = 0;
 				size_t len = MAXPATHLEN * 2;
 				nvpair_t *elem = NULL;
+				char **errl = NULL, *pathbuf = NULL;
+				nvlist_t **errnvl = NULL;
 
 				for (nvpair_t *pair =
 				    nvlist_next_nvpair(nverrlist, NULL);
 				    pair != NULL;
 				    pair = nvlist_next_nvpair(nverrlist, pair))
 					count++;
-				char **errl = (char **)malloc(
-				    count * sizeof (char *));
+				if (cb->cb_verbosity < 2)
+					errl = (char **)malloc(
+					    count * sizeof (char *));
+				else {
+					pathbuf = safe_malloc(len);
+					errnvl = calloc(count,
+					    sizeof (nvlist_t *));
```

Comment:
3 different allocation functions in 6 lines looks like a bit too much of diversity.  Also one is "safe", while other two are not, while I see no any checks later.